modules/oe_translation_remote/src/Form/RemoteTranslationReviewForm.php

```diff
@@ -192,6 +199,25 @@ public function synchronise(array &$form, FormStateInterface $form_state): void
     $form_state->setRedirect('entity.' . $entity->getEntityTypeId() . '.remote_translation', [$entity->getEntityTypeId() => $entity->id()]);
   }
 
+  /**
+   * Redirects the user to the preview path of the translation request.
+   *
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function preview(array &$form, FormStateInterface $form_state) {
+    /** @var \Drupal\oe_translation_local\TranslationRequestLocal $translation_request */
+    $translation_request = $this->entity;
+
+    $language = $form_state->get('language');
+    $url = $translation_request->toUrl('preview');
+    $url->setRouteParameter('language', $language->id());
```

Comment:
Is this needed? You are setting the URL option the language and in the controller you are checking the current language, not the route parameter.

Reply:
Yes, the language is a required parameter for the route. I removed the option and the check for the current language in controller.